crates/ethereum/primitives/src/transaction.rs

```diff
@@ -750,81 +750,89 @@ impl reth_codecs::Compact for TransactionSigned {
     }
 }
 
-impl FillTxEnv for TransactionSigned {
+impl FillTxEnv<TxEnv> for TransactionSigned {
     fn fill_tx_env(&self, tx_env: &mut TxEnv, sender: Address) {
```

Comment:
I think we should reconsider this now and remove the fill function and instead do `-> TxEnv`